### PR TITLE
feat: Add support for new ignoreMissing and ignoreOptions options

### PR DESCRIPTION
**Pull request by kamilogorek**
- update `sentry-cli` to `1.67.2`
- add support for new `stripCommonPrefix`, `ignoreMissing` and `ignoreOptions` options
- rename `validate.ts` to `options.ts` as its actually fetching options, not only validating them
- refactor `getShouldFinalize` to a generic `getBooleanOption`
- remove custom TS types and use real ones from `sentry-cli`

Fixes https://github.com/getsentry/action-release/issues/41
Fixes https://github.com/getsentry/action-release/issues/59
Fixes https://github.com/getsentry/action-release/issues/53

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,13 +48,16 @@ The following are all _required_.
 |---|---|---|
 |`environment`|Set the environment for this release. E.g. "production" or "staging". Omit to skip adding deploy to release.|-|
 |`finalize`|When false, omit marking the release as finalized and released.|`true`|
+|`ignore_missing`|When the flag is set and the previous release commit was not found in the repository, will create a release with the default commits count instead of failing the command.|`false`|
+|`ignore_empty`|When the flag is set, command will not fail and just exit silently if no new commits for a given release have been found.|`false`|
```
Comment (AbhiPrasad) [inline README.md:52]: Feels bad to duplicate this between the README and `action.yml`, but I realized that there are slight differences between the readme and this (for ex, `version` defaults to `{{github.sha}}`, but there is not default field in yml for that key).

File: src/options.ts [hunk 2]
```diff
@@ -82,17 +82,22 @@ export const getSourcemaps = (): string[] => {
 };
 
 /**
- * Find out from input if we should finalize the release.
+ * Find out from input if we should perform an action.
```
Comment (AbhiPrasad) [inline src/options.ts:85]: What do you think about this instead?

```suggestion
 * Fetch boolean option from input. Throws error if option value is not a boolean.
```

File: action.yml [hunk 3]
```diff
@@ -11,6 +11,12 @@ inputs:
   finalize:
     description: 'When false, omit marking the release as finalized and released.'
     default: true
+  ignore_missing:
+    description: 'When the flag is set and the previous release commit was not found in the repository, will create a release with the default commits count instead of failing the command.'
+    required: false
```
Comment (AbhiPrasad) [inline action.yml:16]: `ignore_missing` and `ignore_empty` default to `false`. Do we have to update the `action.yml` to add a `default: true` for these fields?

**Review by AbhiPrasad — COMMENTED** [review]
✅ but will leave final approval rights to Ecosystem team.

File: action.yml (hunk 3 above)
Comment (kamilogorek) [reply action.yml:16]: We don't, as we do that here https://github.com/getsentry/action-release/pull/63/files#diff-4fab5baaca5c14d2de62d8d2fceef376ddddcc8e9509d86cfa5643f51b89ce3dR15-R16

**Review by mgaeta — APPROVED** [review]
Looks great 👍🏽

**Review by scefali — APPROVED** [review]

**Comment by kamilogorek** [comment]
@scefali @mgaeta could one of you perform a  new release please? Thanks

**Pull request merged**